core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -602,6 +606,19 @@ public CallTracer create() {
     channelz.addRootChannel(this);
   }
 
+  private static Attributes addProxyToAttributes(ProxyDetector proxyDetector,
+      Attributes attributes) {
+    if (attributes == null) {
```

Comment:
Nit: won't ever be null, so remove. And if it was null, I'd sort of be happy if it crashed so we could see where the faulty value was coming from (but there's a checkNotNull already in place).

Reply:
done